lib/ansible/runner.rb

```diff
@@ -383,7 +379,7 @@ def wait_for(path, timeout: 10.seconds)
 
         begin
           res = yield
-          raise "Timed out waiting for #{path}" unless path_created.wait(timeout)
+          raise "Timed out waiting for #{target_path}" unless path_created.wait(timeout)
```

Comment:
Interesting when the command fails this "times out".  Think we should use `AwesomeSpawn.run!` so it would raise an exception rather than wait here

Reply:
Since the run is async, I don't think so?  But I agree that if there are higher order errors like ansible-runner not existing, we should be able to handle those.

Reply:
Or does the ansible-runner start command return immediately? (Still not sure of the point of the wait_for)

Reply:
This only happens because of the psutil version mismatch though so we can look at this in a follow-up I think

Reply:
> Since the run is async, I don't think so? But I agree that if there are higher order errors like ansible-runner not existing, we should be able to handle those.

Right this is at the level of "if ansible-runner is not found we still wait for the directory"

> Or does the ansible-runner start command return immediately? (Still not sure of the point of the wait_for)

I believe the problem was that ansible-runner start returns right away, but you can't check if it is done right away.  And polling opened up the issue if it exited too quickly.

Reply:
https://github.com/ManageIQ/manageiq/pull/20666#issue-716658194 has a good description of the issue, but I'm curious when they stopped writing a `pid` file